### ISSUE TITLE
Reconsider mapping 4X, 6X and 7X conditions as "rainy"

### ISSUE THREAD

**Issue by nadiamoe** [Noltari/AEMET-OpenData#8]
Heyo!

I'm a homeassistant user, and I arrived to this repo after getting a bit in the rabbit hole of trying to figure out why aemet was "lying" to me.

Bit of background first: In my household, it has become a bit of a meme how AEMET "never gets the forecast right", specially regarding to rain. We joke saying that whenever AEMET says it's going to rain, it is definitely safe to assume it won't. It has gotten to absurd points where it says it is currently rainy while the sky is blue and the sun is shining.

This is funny as an inside joke but also slightly frustrating as it makes the forecast a bit... welp, useless. So I went to try and find _why_ the forecast was routinely wrong regarding to rain so often. So today being one of those days where the sky is blue and not a drop of water can be seen, I decided to go ahead and look at some data to see if I understood what's happening.

Here's what aemet openData has to say about today's weather:

```xml
<estado_cielo periodo="20" descripcion="Cubierto con lluvia escasa">46</estado_cielo>
<estado_cielo periodo="21" descripcion="Cubierto con lluvia escasa">46n</estado_cielo>
<estado_cielo periodo="22" descripcion="Cubierto con lluvia escasa">46n</estado_cielo>
```

Now, if I scroll a bit down to get the detail for those, I see:
```xml
<precipitacion periodo="20">0.1</precipitacion>
<precipitacion periodo="21">0.1</precipitacion>
<precipitacion periodo="22">0.1</precipitacion>
<precipitacion periodo="23">0.1</precipitacion>
```

And
```xml
<prob_precipitacion periodo="2002">65</prob_precipitacion>
```

In light of this I think it now makes sense. AEMET forecast is not wrong, we definitely do not have the same concept for what "lluvia escasa" (scant rain) means. I'm not sure if AEMET documents what do they quantitatively mean by "escasa", but at least in my experience, "lluvia escasa" means "no rain".

This takes me to the title of the issue: At least in my experience, I think conditions that AEMET considers "escasas" are better mapped to their no-rain versions than to their rain versions. E.g. I think it would be more representative to map `4X` conditions to `AOD_COND_CLOUDY` than to `AOD_COND_RAINY`.

Curious to hear your thoughts on this!

**Comment by Noltari** [comment]
The issue here is that Home Assistant lacks the "scant" conditions in its weather condition mapping:
https://www.home-assistant.io/integrations/weather/#condition-mapping

In order to map AEMET conditions to Home Assistant I checked the icons provided for each condition and I discovered that the difference between the "normal" and "scant" values is the amount of water/snow depicted in the images.

See the following examples:

23 (Intervalos nubosos con lluvia) -> https://www.aemet.es/imagenes/png/estado_cielo/23_g.png
43 (Intervalos nubosos con lluvia escasa) -> https://www.aemet.es/imagenes/png/estado_cielo/43_g.png

36 (Cubierto con nieve) -> https://www.aemet.es/imagenes/png/estado_cielo/36_g.png
74 (Cubierto con nieve escasa) -> https://www.aemet.es/imagenes/png/estado_cielo/74_g.png

So the only difference I can see is the amount of snow or rain...
Usually if a higher amount of rain/snow is forecasted, then the probability of raining/snowing is obviously higher.
Therefore, being able to see the amount of precipitation and its probability in Home Assistant adds the remaining context to the condition. With these 3 values (condition, precipitation quantity and precipitation probability) you're also able to predict if there's a high or low chance of raining/snowing.

For me, it's perfectly fine the way it is right now due to the current Home Assistant limitations, but I will happily change it in the future if new conditions are added to the Home Assistant weather conditions.

**Comment by nadiamoe** [comment]
> So the only difference I can see is the amount of snow or rain...
Usually if a higher amount of rain/snow is forecasted, then the probability of raining/snowing is obviously higher.

This seems to be the case and matches my observations as well.

> Therefore, being able to see the amount of precipitation and its probability in Home Assistant adds the remaining context to the condition.

I also agree with this, but I think there's an important factor to be considered too: Homeassitant does not make those values (amount and probability) visible anywhere!

See the default weather card:
![image](https://github.com/Noltari/AEMET-OpenData/assets/969721/00f2952f-acde-483a-b34e-83f770c0f19b)

And also the entity view:
![image](https://github.com/Noltari/AEMET-OpenData/assets/969721/d3a2714f-aee1-4fe6-af93-905cde9d1776)

Basically I need to manually call `weather.get_forecasts` if I want to see those values. I think it's fair to say that homeassisant is relying too much on the condition, and too few on the values to actually surface this information. Which we could argue Homeassistant could do a better job to surface this information, but I think that's going to be harder to change (because it's a hard problem to solve).

My hypothesis is this: AEMET's definition of "scant" is very close to zero, and the moment it becomes significant, AEMET will report the non-scant version. What are your thoughts about this? If we had the real numbers AEMET uses and those were, for example, that "scant" means less than 1mm of precipitation, would you be open to make this change?

I think I can spend a couple hours doing some datamining to get the numbers if that can help :)

**Comment by Noltari** [comment]
> I also agree with this, but I think there's an important factor to be considered too: Homeassitant does not make those values (amount and probability) visible anywhere!
> 
> 
> 
> See the default weather card:
> 
> ![image](https://github.com/Noltari/AEMET-OpenData/assets/969721/00f2952f-acde-483a-b34e-83f770c0f19b)
> 
> 
> 
> And also the entity view:
> 
> ![image](https://github.com/Noltari/AEMET-OpenData/assets/969721/d3a2714f-aee1-4fe6-af93-905cde9d1776)

You're right, the default Home Assistant card doesn't show the amount and probability, but there are third party cards that show them.

This is my weather setup:
![image](https://github.com/Noltari/AEMET-OpenData/assets/264346/24797ce4-1203-45c9-be27-6ae970055bba)

https://github.com/decompil3d/lovelace-hourly-weather
https://github.com/pkissling/clock-weather-card

**Comment by nadiamoe** [comment]
Oh, I had https://github.com/pkissling/clock-weather-card but not https://github.com/decompil3d/lovelace-hourly-weather! Will give that one a try 👀
